python/cudf_polars/tests/dsl/test_expr.py

```diff
@@ -73,3 +73,26 @@ def test_namedexpr_repr_stable():
     b2 = expr.NamedExpr("b1", expr.Col(plc.DataType(plc.TypeId.INT8), "a"))
 
     assert repr(b1) == repr(b2)
+
+
+def test_equality_cse():
+    dt = plc.DataType(plc.TypeId.INT8)
+
+    def make_expr(n1, n2):
+        a = expr.Col(plc.DataType(plc.TypeId.INT8), n1)
+        b = expr.Col(plc.DataType(plc.TypeId.INT8), n2)
+
+        return expr.BinOp(dt, plc.binaryop.BinaryOperator.ADD, a, b)
+
+    e1 = make_expr("a", "b")
+    e2 = make_expr("a", "b")
+    e3 = make_expr("a", "c")
+
+    assert e1.children is not e2.children
+    assert e1 == e2
+    assert e1.children is e2.children
+    assert e1 == e1
+    assert e2 == e2
+    assert e1 != e3
+    assert e2 != e3
+    assert e3 == e3
```

Comment:
I'm not sure exactly what cases you want to test. Should we verify that `e1 == e1` and `e2 == e2` both before and after the CSE is elided? Does the `e3 == e3` check add anything that the other checks don't already do?

Reply:
It seems like there are some extra checks here but it's not a big deal. Feel free to address as you see fit.

Reply:
Tidied this a bit.